src/pip/_internal/configuration.py

```diff
@@ -222,7 +221,7 @@ def unset_value(self, key):
                 # name removed from parser, section may now be empty
                 section_iter = iter(parser.items(section))
                 try:
-                    val = next(section_iter)
+                    val = next(section_iter)  # type: Optional[Tuple[str, str]]
                 except StopIteration:
                     val = None
```

Comment:
Looking at [the type hint of RawConfigParser.items](https://github.com/python/typeshed/blob/6028226750af927e1d728788480c15335d67c26b/stdlib/3/configparser.pyi#L133-L134), it seems that we can drop this and use `if not parser.items(section)` in the next statement.

Reply:
I wasn't able to fully understand your comment, but I have added https://github.com/pypa/pip/pull/8566/commits/4fb7c6c1148855eec96d2f4da1feb47f11d8acbd based on what I could. Please take a look at the same, and let me know if this is what you meant 😊

Reply:
Yes, that's what I meant!  BTW 4fb7c6c left a there's a trailing newline.

Reply:
Good catch, fixed in https://github.com/pypa/pip/pull/8566/commits/72f075cd25a102047149529e81ec418b929047ee

Reply:
My suggestion seems to make some tests fail on Python 2 on non-Windows OSes.  I can't wrap my mind around the reason because [the return type of RawConfigParser.items on Python 2 is also a list](https://github.com/python/typeshed/blob/209b6bb127f61fe173a60776e23883ac450cf1c8/stdlib/2/ConfigParser.pyi#L71).

Reply:
I saw the CI, and the svn test (`tests/functional/test_uninstall.py::test_uninstall_editable_from_svn`) is failing ([here](https://dev.azure.com/pypa/pip/_build/results?buildId=26754&view=logs&j=f801bdd4-2ff4-5da8-3bd6-deea013b4470&t=ada738ba-3760-52e8-582e-eef0693a80cb&l=147) and [here](https://dev.azure.com/pypa/pip/_build/results?buildId=26754&view=logs&j=f801bdd4-2ff4-5da8-3bd6-deea013b4470&t=ada738ba-3760-52e8-582e-eef0693a80cb&l=147))

Were you talking about those failures?

Reply:
Yep, I think it'd be fair trying to rerun it, just in case it's flaky by e.g. squashing the commits.

Reply:
It was flaky indeed. Squashing and re-running passes all the tests.

Reply:
I have reverted the above suggested changed as per @uranusjr comment https://github.com/pypa/pip/pull/8566#discussion_r453036309

Reply:
@uranusjr, just in case you missed the discussion here, `RawConfigParser.items` always return a list, not only hinted in typeshed but also in CPython docs ([Python 2](https://docs.python.org/2/library/configparser.html#ConfigParser.RawConfigParser.items), [Python 3](https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.items)), and CPython has a reputation of rarely ever break backward compatibility.  I'm following this up in GH-8571.